tests/common/mod.rs

```diff
@@ -267,6 +267,14 @@ impl TestEnvironment {
             })
             .to_string()
     }
+
+    /// Used before mutating operations to create more predictable commit ids
+    /// and change ids in tests
+    pub fn advance_test_rng_seed_to_multiple_of(&self, step: i64) {
+        assert!(step > 0, "step must be >0, got {step}");
+        let mut command_number = self.command_number.borrow_mut();
+        *command_number = step * (*command_number / step) + step;
+    }
```

Comment:
I didn't want people to have to worry too much about adding too many commands in between two calls to this function. I feel like people should be encouraged to modify tests, so I wanted to create as little additional friction as possible.

You're certainly right that there's a cost to that in the behavior becoming less predictable. I'm OK with tests only being somewhat predictable.

Reply:
Well, but if the alignment isn't large enough, the command_number will change by inserting commands inbetween. I think it's better to panic in that case.

Reply:
Usually, that just means some insta tests change (or panic, if you aren't using insta).

I was thinking that alignment of ~20 should be good for most tests (and for test sections, even more so). The idea is that you use 200_000 at first, then if you want to insert a "subheading" you could use 10_000 (though inserting another 200_000 step and updating the change ids is also OK), then 500, then 25 (though at that point the test should probably be split into several). I was also thinking of adding such a guideline to the docstring.

Reply:
I guess the thing to worry about is that somebody might think that the commit ids staying the same is "guaranteed" in some strong sense and be surprised. But this only happens when they are already editing the test.

Reply:
> that just means some insta tests change

Ok, so stabilizing commit/change ids is soft requirement.

> The idea is that you use 200_000 at first, then if you want to insert a "subheading" you could use 10_000 (though inserting another 200_000 step and updating the change ids is also OK), then 500, then 25 

I feel it would be messier than increasing absolute numbers (like 10_000, 15_000, ...)
Do we need support sub-heading insertion? If not, I think a bump function with no parameter is good enough. I don't have strong opinion, though.

Reply:
I think I'll keep this for now. If it turns out to be a mistake, it won't be very serious, we can always just remove all calls to this function and be back at where we were.